src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -192,7 +194,7 @@ class MutexLockerImpl: public StackObj {
   MutexLockerImpl(Mutex* mutex, Mutex::SafepointCheckFlag flag = Mutex::_safepoint_check_flag) :
     _mutex(mutex) {
     bool no_safepoint_check = flag == Mutex::_no_safepoint_check_flag;
-    if (_mutex != nullptr) {
+    if (_mutex != nullptr && Thread::current_or_null() != nullptr) {
```

Comment:
I do not like this addition to the mutex locker code. First, it probably needs to be `Thread::current_or_null_safe` in case we ever lock in a signal-handling context. But second this slows down all mutex uses. Instead `NMTLocker` should extend ConditionalMutexLocker and make the current thread check the condition.

Reply:
Ok I've taken those checks out of `MutexLockerImpl` and applied your suggestion.

Reply:
It may be that not all uses were switched over to ConditionalMutexLocker when it was introduced. Passing null during init phase was common practice when no locking needed; possibly also cases of passing null if already locked.